keps/sig-instrumentation/4785-resource-state-metrics/README.md

```diff
@@ -0,0 +1,1170 @@
+removing all managed resources will not remove the controller.
+
+The reason why it is so is because cluster-scoped resources cannot have owner
+references linking back to namespace-scoped resources, in which case, the
+garbage collection is a no-op. See [819a80] for more details.
+
+[819a80]: https://github.com/rexagod/resource-state-metrics/commit/819a8001200a13c51cb82779c139a9081b4d613b
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+In the context of the controller, pre-existing managed resources in the cluster
+will be picked back up after a (re-)deploy.
+
+###### Are there any tests for feature enablement/disablement?
+
+<!--
+The e2e framework does not currently support enabling or disabling feature
+gates. However, unit tests in each component dealing with managing data, created
+with and without the feature, are necessary. At the very least, think about
+conversion tests if API types are being modified.
+
+Additionally, for features that are introducing a new API field, unit tests that
+are exercising the `switch` of feature gate itself (what happens if I disable a
+feature gate after having objects written with the new field) are also critical.
+You can take a look at one potential example of such test in:
+https://github.com/kubernetes/kubernetes/pull/97058/files#diff-7826f7adbc1996a05ab52e3f5f02429e94b68ce6bce0dc534d1be636154fded3R246-R282
+-->
+
+N/A.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+Future API version for managed resources will follow the [hub-spoke] 
+interconversion model.
+
+[hub-spoke]: https://www.kubebuilder.io/multiversion-tutorial/conversion-concepts
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+In the context of the controller, failed controller deployments do not make any
+changes to the existing managed resources' metric configurations.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+The controller has a telemetry server to diagnose and query statistics.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+Yes, controller deploy, scale down, and scale up works fine. Operations on
```

Comment:
Please either give more details how exactly it was tested (see https://github.com/kubernetes/enhancements/pull/4702/files#diff-a4947474bee8c14f0b3ffb9ecdc0a12f26c4894b3d89657ea06c51a0f79e6fb2R535-R607 as an exampl)

or remove this answer and replace with TODO - it's not yet required for Alpha.